database/api.go

```diff
@@ -25,19 +26,19 @@ type API interface {
 	// CommitOrRollback will rollback on an errors.TracerError otherwise commit
 	CommitOrRollback(err error) errors.TracerError
 
-	// Count the number of rows in the passed query
-	Count(qb.Table, *qb.SelectQuery) (int32, error)
 	// CountWhere rows match the passed condition in the specified table. Condition
 	// may be nil in order to just count the table rows.
 	CountWhere(qb.Table, *qb.ConditionExpression) (int32, error)
 	// Create initializes a Record and inserts it into the Database
 	Create(obj record.Record) errors.TracerError
+	// Count the number of rows in the passed query
+	Count(qb.Table, *qb.SelectQuery) (int32, error)
+	// Select executes a given select query and populates the target
+	Select(target interface{}, query *qb.SelectQuery, options *record.ListOptions) errors.TracerError
```

Comment:
Why did these move?

Reply:
I subclassed the interface to have these but then decided not to and put them back. I didn't remember the order. I will move them back.